oemof/solph/blocks.py

```diff
@@ -355,23 +378,47 @@ def _create(self, group=None):
         def _investvar_bound_rule(block, i, o):
             """Rule definition for bounds of invest variable.
             """
-            return (m.flows[i, o].investment.minimum,
-                    m.flows[i, o].investment.maximum)
-        # create variable bounded for flows with investement attribute
-        self.invest = Var(self.FLOWS, within=NonNegativeReals,
+            if (i, o) in self.CONVEX_INVESTFLOWS:
+                return (m.flows[i, o].investment.minimum,
+                        m.flows[i, o].investment.maximum)
+            elif (i, o) in self.NON_CONVEX_INVESTFLOWS:
```

Comment:
Is this really necessary? Below, there is the _max_invest_rule which already sets the upper bound for the nonconvex case. Are do we need some bounds here?

Reply:
for now, we need the definition of the bounds